parlai/tasks/coqa/build.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+# Download and build the data if it does not exist.
+
+import parlai.core.build_data as build_data
+import os
+import json
+
+tfname = 'coqa-train-v1.0.json'
+dfname = 'coqa-dev-v1.0.json'
+url = 'https://nlp.stanford.edu/data/coqa/'
+
+
+def make_parlai_format(outpath, dtype, data):
+    print('building parlai:' + dtype)
+    fout = open(os.path.join(outpath, dtype + '.txt'), 'w')
+    for each in data:
+        output = ""
```

Comment:
again, it's better (read: more efficient) to use arrays with a StringBuilder type mechanic.

```python
output = []
output.append("item1")
output.append("item2")
fout.write("\n".join(output))
```